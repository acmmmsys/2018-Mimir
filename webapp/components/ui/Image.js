import React, { Component } from 'react';
import classnames from 'classnames';

import 'style/ui/Image.scss';

const defaultClassNames = [
    'ui-image',
];

class Image extends Component {

    static defaultProps = {
        source: '',
        label: '',
        width: 100,
        height: 100,
        classNames: [],
        clickable: false,
        attached: false,
        selected: false,
        loading: false,
        onClick: Function.prototype,
    }

    render() {
        const classNames = classnames(
            defaultClassNames,
            this.props.classNames,
            {
                'ui-image-clickable': this.props.clickable,
                'ui-image-highlight-attached': this.props.attached,
                'ui-image-highlight-selected': this.props.selected,
            }
        );

        if (this.props.loading) {
            return (
                <div className={classNames} onClick={this.props.onClick}>
                    <img src={this.props.src}
                        width={this.props.width}
                        height={this.props.height}/>
                    <div className='ui active dimmer'>
                        <div className='ui loader'/>
                    </div>
                </div>
            );
        }
        
        return (
            <div className={classNames} onClick={this.props.onClick}>
                <img src={this.props.src}
                     width={this.props.width}
                     height={this.props.height}/>
                {this.props.label ?
                    <span className='ui-image-label-overlay'>
                        <span className='ui-image-label'>
                            {this.props.label}
                        </span>
                    </span>
                : ''}
                
            </div>
        );
    }
}

export default Image;