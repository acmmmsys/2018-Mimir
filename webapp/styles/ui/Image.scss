.ui-image {
    position: relative;
}

.ui-image-label-overlay {
    position: absolute;
    top: 70px;
    left: 0;
    width: 100%;
}

.ui-image-spinner-overlay {
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    background: rgb(0, 0, 0);
    background: rgba(0, 0, 0, 0.7);
}

.ui-image-spinner {
    position: absolute;
    width: 3px;
    height: 30px;
    top: 30px;
    left: 30px;
    border-radius: 50%;
    border: 16px solid #000;
    border-color: #ffffff transparent #ffffff transparent;
    -webkit-animation: loading 1.3s linear infinite;
    animation: loading 1.3s linear infinite;
    color: white;
    box-sizing: border-box;
}

.ui-image-label {
    color: white;
    letter-spacing: -1px;
    background: rgb(0, 0, 0);
    background: rgba(0, 0, 0, 0.7);
    padding: 5px;
    box-sizing: border-box;
}

.ui-image-clickable {
    cursor: pointer;
}

.ui-image-highlight {
    border: solid yellow 1px;
}

.ui-image-highlight-selected {
    border: solid white 1px;
}

.ui-image-highlight-loading {
    border: solid green 1px;
}

.ui-image-highlight-attached {
    border: solid yellow 1px;
}

@-webkit-keyframes loading {
    0% {
      -webkit-transform: rotate(0);
      transform: rotate(0);
    }
    100% {
      -webkit-transform: rotate(360deg);
      transform: rotate(360deg);
    }
  }
  @keyframes loading {
    0% {
      -webkit-transform: rotate(0);
      transform: rotate(0);
    }
    100% {
      -webkit-transform: rotate(360deg);
      transform: rotate(360deg);
    }
}