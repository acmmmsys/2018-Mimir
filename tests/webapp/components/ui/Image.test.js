import React from 'react';
import Image from '../../../../webapp/components/ui/Image'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
 
const mockStore = configureStore();

let store;
let wrapper;

test('Image does render', () => {
  const wrapper = shallow(
      <Image store={mockStore(initialState)}>This is a test!</Image>
  );
  expect(wrapper).toMatchSnapshot();
});
