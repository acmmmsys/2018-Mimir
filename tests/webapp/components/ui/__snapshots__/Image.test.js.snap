// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image does render 1`] = `
<div
  className="ui-image "
  onClick={[Function]}
>
  <img
    height={100}
    width={100}
  />
</div>
`;
